server/log_rpc_server.go

```diff
@@ -91,6 +91,13 @@ func (t *TrillianLogRPCServer) QueueLeaves(ctx context.Context, req *trillian.Qu
 	err = tx.QueueLeaves(req.Leaves, t.timeSource.Now())
 	if err != nil {
 		tx.Rollback()
+
+		if se, ok := err.(storage.Error); ok {
+			if se.ErrType == storage.DuplicateLeaf {
```

Comment:
make this a type switch (with a `default` so newly-created storage errors won't get silently lost)